qcodes/instrument_drivers/tektronix/Keithley_3706A_Matrix_channels.py

```diff
@@ -0,0 +1,600 @@
+import itertools
+from typing import List, Dict, Optional, Union, Tuple
+
+import qcodes.utils.validators as vals
+from qcodes import VisaInstrument
+from qcodes.utils.helpers import create_on_off_val_mapping
+
+
+class UnknownOrEmptySlot(Exception):
+    pass
+
+
+class InvalidValue(Exception):
+    pass
+
+
+class Keithley_3706A(VisaInstrument):
+    """
+    This is the QCoDeS instrument driver for the Keithley 3706A-SNFP
+    System Switch.
+    """
+
+    def __init__(self, name: str, address: str, **kwargs) -> None:
+        """
+        Args:
+            name: Name to use internally in QCoDeS
+            address: VISA resource address
+        """
+        super().__init__(name, address, terminator='\n', **kwargs)
+
+        self.add_parameter('reset_channel',
+                           get_cmd=None,
+                           set_cmd=self._reset_channel,
+                           docstring='Resets the specified channels to '
+                                     'factory default settings.')
+
+        self.add_parameter('open_channel',
+                           get_cmd=None,
+                           set_cmd=self._open_channel,
+                           docstring='Opens the specified channels and '
+                                     'backplane relays.')
+
+        self.add_parameter('close_channel',
```

Comment:
I tested the set command with this function on the backplane relays, and it does not raise an exception if the interlock for that backplane relay is disabled.  It would probably be a good idea to raise this exception or at least issue a warning so that the user knows that the relay is not in the desired state.